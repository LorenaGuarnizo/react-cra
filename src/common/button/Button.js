import React, { Component } from 'react';
import { buttonPropTypes } from './Button.props';

import './Button.scss';

class Button extends Component {
    render() {
      return (
        <div className="button-container">
          <header className="button-header">
            <h1 className="app-title">Click on button</h1>
          </header>
          <button type="button" onClick={this.props.countAction}>Click</button>
          <p>Counter: {this.props.counter}</p>
        </div>
      );
    }
  
    static propTypes = buttonPropTypes;
  }

  export default Button;
  