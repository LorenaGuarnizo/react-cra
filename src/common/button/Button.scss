.button-container {
    margin: 0 auto;
    max-width: 450px;
    width: 100%;
    .text-value,
    button {
        text-align: center;
        color: #404040;
        font: 100 1em/150% "proxima-nova", Helvetica, sans-serif;
        font-size: 28px;
    }
    button {
        cursor: pointer;
    }
}