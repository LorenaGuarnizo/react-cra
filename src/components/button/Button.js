import React, { Component } from 'react';
import { connect } from 'react-redux';

import { simpleAction } from './Button.actions'
import { buttonPropTypes } from './Button.props';

import './Button.scss';

class Button extends Component {
  render() {
    return (
      <div className="button-container">
        <header className="button-header">
          <h1 className="App-title">Click on button</h1>
        </header>
        <button type="button" onClick={this.props.simpleAction}>Click</button>
        <p>{this.props.counter}</p>
      </div>
    );
  }

  static propTypes = buttonPropTypes;
}

const mapStateToProps = state => {
  const { counter } = state.button;
  return { counter };
};

export default connect(mapStateToProps, { simpleAction })(Button);