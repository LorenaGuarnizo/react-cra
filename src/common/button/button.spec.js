import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Button, {
  ERROR_MESSAGE_GEATERTHAN,
  ERROR_MESSAGE_LESSTHAN
} from "./Button";

describe("Common Counter Component", () => {
  const props = {
    setIncrement: jest.fn(),
    setDecrement: jest.fn()
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should show an error message when the counter number is greater than 9", () => {
    const wrapper = shallow(<Button {...props} counter={10} />);
    const incrementButton = wrapper.find("button#increment");
    incrementButton.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual(ERROR_MESSAGE_GEATERTHAN);
  });

  it("should show an error message when the counter number is less than 0", () => {
    const wrapper = shallow(<Button {...props} counter={0} />);
    const incrementButton = wrapper.find("button#decrement");
    incrementButton.simulate("click");
    const text = wrapper.find("p").text();
    expect(text).toEqual(ERROR_MESSAGE_LESSTHAN);
  });

  it("should count by 1 when the increment button is clicked", () => {
    const wrapper = shallow(<Button {...props} counter={0} />);
    const incrementButton = wrapper.find("button#increment");
    incrementButton.simulate("click");
    const text = wrapper.find("input#counterValue").props().value;
    expect(text).toEqual(1);
  });

  it("should decrement by 0 when the decremet button is clicked", () => {
    const wrapper = shallow(<Button {...props} counter={1} />);
    const incrementButton = wrapper.find("button#decrement");
    incrementButton.simulate("click");
    const text = wrapper.find("input#counterValue").props().value;
    expect(text).toEqual(0);
  });
});
