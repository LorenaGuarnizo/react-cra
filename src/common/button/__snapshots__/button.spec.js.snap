// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Counter Component renders correctly 1`] = `
<div
  className="button-container"
>
  <header
    className="button-header"
  >
    <h1
      className="app-title"
    >
      Click on button
    </h1>
  </header>
  <button
    id="decrement"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <input
    className="text-value"
    id="counterValue"
    readOnly={true}
    type="text"
    value={0}
  />
  <button
    id="increment"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <p>
    
  </p>
</div>
`;
